### DISCUSSION TITLE
Cant figure out navigation

### DISCUSSION THREAD

**Discussion by vynxc** [unoplatform/uno#9968] [Q&A] [answered]
I'm making a streaming app and want to bring up a new episode page. I want this page to appear on top of everything else and be close able with a button. the INavigator doesn't exist and i cant find anything else. This is my first mobile application and I'm not using mvvm. the repo is  https://github.com/vynxcc/AnimeApp . Under ShowControl I have a MyGrid_Tapped method where I would like navigate to a new Episode Page that displays ontop of everything.

**Comment by vynxc** [comment]
well actually second app i made the same app in xamarin

**Comment by nickrandolph** [comment, marked as answer]
In order to Navigate to a new page you need a reference to a Frame. Since the ShowControl is being created by the Home control which is being shown in the MainPage, you'll need to traverse up the visual tree (you can use VisualTreeHelper.GetParent) until you get to MainPage, which has a Frame property.
I would suggest that you can do more of the code you have in the codebehind files in XAML and data binding, which would simplify the layout of your application significantly.

**Reply by vynxc** [reply]
All I really have is properties and requests. But besides that's could you provide an example of how I would use VisualTreeHelper. I don't think I understand what you mean. Where should the frame be?

**Reply by vynxc** [reply]
And after that. How would I close the page

**Reply by vynxc** [reply]
I was hoping to do something like this await Navigation.PushModalAsync(view); like in xamarin. Is that not possible?

**Reply by nickrandolph** [reply]
Frame.Navigation is supported on all the Uno platforms.
The issue you have is that you ShowControl is midway down the visual tree, so you need to be able to get a reference to the Frame, which is a property on MainPage.
The other alternative is to expose the _window.Content as a property on App ie in App.xaml.cs you could have

public Frame Frame => _window.Content as Frame;

Then you can access this anywhere eg
(Application.Current as App).Frame.Navigate(....)

**Reply by vynxc** [reply]
> Frame.Navigation is supported on all the Uno platforms. The issue you have is that you ShowControl is midway down the visual tree, so you need to be able to get a reference to the Frame, which is a property on MainPage. The other alternative is to expose the _window.Content as a property on App ie in App.xaml.cs you could have
> 
> public Frame Frame => _window.Content as Frame;
> 
> Then you can access this anywhere eg (Application.Current as App).Frame.Navigate(....)

Thanks so much man. I'll test it out later. Means alot 😁

**Reply by vynxc** [reply]
So I've tried public Frame Frame => _window.Content as Frame; with (Application.Current as App).Frame.Navigate(typeof(Episode_View)); and nothing happens. But the click is registered. Not exclusively sure how to use VisualTreeHelper.GetParent
